substrate/pkg/controller/substrate/cluster/addons/tekton.go

```diff
@@ -43,6 +162,9 @@ func (t *Tekton) Create(ctx context.Context, substrate *v1alpha1.Substrate) (rec
 			return reconcile.Result{}, fmt.Errorf("applying tekton, %w", err)
 		}
 	}
+	if err := client.ApplyYAML(ctx, []byte(rbac)); err != nil {
```

Comment:
Why do we need to apply this? Is the tekton release broken?

Reply:
The tekton release yaml files defines the minimum RBAC rules for it to run. However, we need a separate RBAC rules for the service account that execute the tasks in the pipeline. Those are usually defined by the Tekton user. Since Tekton now is part of KIT as add-on, it should come with the permissions to deploy control and data planes. These are the rules that it needs for the current pipeline. We will adjust it if in the future we change the control plane and data plane deployment tasks.